File: src/main/java/com/micahelias/opengl/IndexBuffer.java
package com.micahelias.opengl;


import static org.lwjgl.opengl.GL15.*;

/**
 * VertexBuffer
 */
public class IndexBuffer {

  private int m_RendererID;
  private int m_Count;
  
  public IndexBuffer(int[] data) {
    m_Count = data.length;
    m_RendererID = glGenBuffers();
    glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, m_RendererID);
    glBufferData(GL_ELEMENT_ARRAY_BUFFER, data, GL_STATIC_DRAW);
  }


  public void bind() {
    glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, m_RendererID);
  }

  public void unbind() {
    glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, 0);
  }

  public int getCount() {
    return m_Count;
  }

}

File: mango/opengl/index_buffer.go
package opengl

import "github.com/go-gl/gl/v3.3-core/gl"

type IndexBuffer struct {
	m_RendererID uint32
	m_Count      int
}

func NewIndexBuffer(data []uint32) *IndexBuffer {
	var ib uint32
	gl.GenBuffers(1, &ib)
	gl.BindBuffer(gl.ELEMENT_ARRAY_BUFFER, ib)
	gl.BufferData(gl.ELEMENT_ARRAY_BUFFER, 4*len(data), gl.Ptr(data), gl.STATIC_DRAW)
	return &IndexBuffer{m_RendererID: ib, m_Count: len(data)}
}

func (ib *IndexBuffer) Bind() {
	gl.BindBuffer(gl.ELEMENT_ARRAY_BUFFER, ib.m_RendererID)
}

func (ib *IndexBuffer) Unbind() {
	gl.BindBuffer(gl.ELEMENT_ARRAY_BUFFER, 0)
}

func (ib *IndexBuffer) GetCount() int {
	return ib.m_Count
}
